File: API/Features/Cashiers/Transactions/Dtos/CashierReadDto.cs
using System;
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Cashiers.Transactions {

    public class CashierReadDto : IMetadata {

        // PK
        public Guid CashierId { get; set; }
        // FKs
        public int CompanyId { get; set; }
        public int SafeId { get; set; }
        // Fields
        public string Date { get; set; }
        public string Entry { get; set; }
        public decimal Amount { get; set; }
        public string Remarks { get; set; }
        public bool IsDeleted { get; set; }
        // Navigation
        public SimpleEntity Company { get; set; }
        public SimpleEntity Safe { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/cashiers/transactions/classes/dtos/cashier-read-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface CashierReadDto extends Metadata {

    date: string
    cashierId: string
    company: SimpleEntity
    safe: SimpleEntity
    entry: string
    amount: number
    remarks: string
    isDeleted: boolean
    postAt: string
    postUser: string
    putAt: string
    putUser: string

}
